neuralpp/test/quick_tests/graphical_model/representation/belief_propagation_test.py

```diff
@@ -0,0 +1,37 @@
+from neuralpp.experiments.experimental_inference.belief_propagation import BeliefPropagation
+from neuralpp.inference.graphical_model.representation.factor.pytorch_table_factor import (
+    PyTorchTableFactor,
+)
+from neuralpp.inference.graphical_model.variable.integer_variable import IntegerVariable
+
+
+def test_belief_propagation():
+    cloudy = [0.2, 0.4, 0.4]
+    sprinkler = [0.6, 0.4]
+    rain = [
+        [0.2, 0.3, 0.5],
+        [0.3, 0.4, 0.3],
+        [0.4, 0.2, 0.4]
+    ]
+
+    def wet_grass(wetness: int, rain_level: int, sprinkler_on: int):
+        return 1.0 if (rain_level + sprinkler_on == wetness) else 0.0
+
+    for log_space in {False, True}:
+        c = IntegerVariable("c", 3)
+        r = IntegerVariable("r", 3)
+        s = IntegerVariable("s", 2)
+        w = IntegerVariable("w", 4)
+
+        model = [
+            PyTorchTableFactor([c], cloudy, log_space=log_space),
```

Comment:
Nice creation of the model! Here we don't need to test for both values of log_space, let's just use the default!